packages/react-core/src/components/ProgressStepper/__tests__/ProgressStep.test.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { ProgressStep, ProgressStepVariant } from '../ProgressStep';
+import { capitalize } from '../../../helpers';
+
+jest.mock('@patternfly/react-icons/dist/esm/icons/check-circle-icon', () => () => 'Success icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/resources-full-icon', () => () => 'Info icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon', () => () => 'Warning icon mock');
+jest.mock('@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon', () => () => 'Danger icon mock');
+
+test('Renders without children', () => {
+  render(
+    <div data-testid="progress-step">
+      <ProgressStep />
+    </div>
+  );
+
+  expect(screen.getByTestId('progress-step').firstChild).toBeVisible();
+});
+
+test('Renders children', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).toBeVisible();
+});
+
+test('Renders with class pf-c-progress-stepper__step on the containing li element', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByRole('listitem')).toHaveClass('pf-c-progress-stepper__step');
+});
+
+test('Renders with custom class name on the containing li element when className prop is provided', () => {
+  render(<ProgressStep className="custom-class">Test</ProgressStep>);
+  expect(screen.getByRole('listitem')).toHaveClass('custom-class');
+});
+
+['success', 'info', 'pending', 'warning', 'danger'].forEach(variant => {
+  test(`Does not render with class pf-m-${variant} by default on containing li element`, () => {
+    render(<ProgressStep>Test</ProgressStep>);
+    expect(screen.getByRole('listitem')).not.toHaveClass(`pf-m-${variant}`);
+  });
+
+  test(`Renders with class pf-m-${variant} when variant = ${variant} on containing li element`, () => {
+    render(<ProgressStep variant={`${variant as ProgressStepVariant}`}>Test</ProgressStep>);
+    expect(screen.getByRole('listitem')).toHaveClass(`pf-m-${variant}`);
+  });
+
+  if (variant !== 'pending') {
+    test(`Does not render with ${variant} variant icon by default`, () => {
+      render(<ProgressStep>Test</ProgressStep>);
+      expect(screen.queryByText(`${capitalize(variant)} icon mock`)).not.toBeInTheDocument();
+    });
+
+    test(`Renders with ${variant} variant icon when variant = ${variant}`, () => {
+      render(<ProgressStep variant={`${variant as ProgressStepVariant}`}>Test</ProgressStep>);
+      expect(screen.getByText(`${capitalize(variant)} icon mock`)).toBeVisible();
+    });
+  }
```

Comment:
@wise-king-sullyman `'pending'`, `'default'`, and `undefined` almost all have the same behavior, however the `pending` variant results in the class `pf-m-pending` being rendered. For this reason, I am not sure if I would be able to combine the 3 into one loop. Perhaps, I could test the `pending` variant separately?

Reply:
@samuelatefah good point, yes I think I'd prefer seeing `pending` tested separately then since as you pointed out it's behavior really isn't fully in line with any of the other variants.